### PR TITLE
feat(prow/config): add support for jenkins jobs in folder

### PR DESCRIPTION
**Pull request by wuhuizuo**
## Why

job in jenkins may be in a folder, not flatten. the  job name for `jenkins` agent  like `folder/job` can not be passed in validation.

### CODE REVIEW

File: prow/config/config.go [hunk 1]
```diff
@@ -2106,12 +2106,30 @@ func (c *Config) validateComponentConfig() error {
...
+		if !jobNmaeRegexJenkins.MatchString(v.Name) {
+			return fmt.Errorf("name: must match regex %q", jobNmaeRegexJenkins.String())
+		}
+	} else {
+		if !jobNameRegex.MatchString(v.Name) {
+			return fmt.Errorf("name: must match regex %q", jobNameRegex.String())
+		}
+	}
+
+	return nil
```
Comment (petr-muller) [inline prow/config/config.go:2125]: You can simplify and deduplicate:

```go
nameRegex := jobNameRegex
if v.Agent == string(prowapi.JenkinsAgent {
	nameRegex = jobNameRegexJenkins
}
if !nameRegex.MatchString(v.Name") {
	return fmt.Errorf("name: must match regex %q", nameRegex.String())
}
return nil
```

File: prow/config/config_test.go [hunk 2]
```diff
@@ -1876,6 +1876,24 @@ func TestValidateJobBase(t *testing.T) {
...
+			name: "valid jenkins job - nested job",
+			base: JobBase{
+				Name:      "folder/job",
+				Agent:     ja,
+				Namespace: &cfg.PodNamespace,
+			},
+			pass: true,
+		},
+		{
+			name: "invalid jenkins job",
```
Comment (petr-muller) [inline prow/config/config_test.go:1889]: Hmm I'm not that familiar with Jenkins anymore, are jobs *always* nested there?

**Comment by wuhuizuo** [comment]
/hold need more commit

When call Jenkins job in folder: `f1/f2/job1`, the Jenkins api path should be `f1/job/f2/job/job1`

**Comment by wuhuizuo** [comment]
I tested with the PR deployment:
image
prow jobs:
https://github.com/PingCAP-QE/ee-ops/blob/main/apps/staging/prow/post/configs/jobs/PingCAP-QE/ee-ops/job-call-jenkins.yaml

**Review by petr-muller — APPROVED** [review]
LGTM, thanks!

**Pull request merged**